packages/client/src/components/TransactionList.js

```diff
@@ -132,52 +62,53 @@ function TransactionDetails({ safeOwners, transaction, onClose }) {
       </div>
     </>
   );
-}
+};
 
-function TransactionListItem({
+const TransactionListItem = ({
   transaction,
   displayIndex,
   isLastInList,
   onView,
-}) {
+}) => {
   const classes = [
-    "is-flex is-align-items-center is-justify-content-space-between py-4 border-light-top",
-    isLastInList && "border-light-bottom",
-  ];
+    "py-4 is-flex is-align-items-center is-justify-content-space-between",
+    isLastInList ? "" : "border-light-bottom",
+  ].join(" ");
   const date = new Date(transaction.eventDate);
-  const lastTokenTransfer =
-    transaction.tokenTransfers[transaction.tokenTransfers.length - 1];
+  const { intent } = transaction.blockEventData.actionView;
+  const status = transaction.flowEventId.includes('ActionDestroyed') 
+    ? 'rejected' : 'confirmed';
 
   return (
-    <div className={classes.join(" ")}>
-      <div className="column">{String(displayIndex + 1).padStart(2, "0")}</div>
-      <div className="column">{lastTokenTransfer?.type ?? "Received"}</div>
-      <div className="column">
-        <TransactionStatusIcon status={transaction.status} />
-        <span className="ml-2">{transaction.status}</span>
-      </div>
-      <div className="column">
-        {getDisplayStringForTokenTransfer(lastTokenTransfer)}
-      </div>
-      <div className="column has-text-grey">
-        {date.toLocaleDateString("en-us")}
-      </div>
-      <div className="column has-text-grey">
-        {date.toLocaleTimeString("en-us")}
-      </div>
-      <div className="column has-text-right">
-        <span
-          className="is-underlined pointer"
-          onClick={() => onView(transaction)}
-        >
-          View
+    <div className={classes}>
+      <Column style={{ maxWidth: maxWidths.index }}>
+        {String(displayIndex).padStart(2, "0")}{' '}
```

Comment:
Can you help me understand what this part does?

Reply:
this code was already here but as I understand it adds a zero to the beginning of the number until it reaches the desired length. This is so we show `01`, `02`, etc instead of `1`, `2,`